parquet-hadoop/src/main/java/org/apache/parquet/hadoop/util/HadoopStreams.java

```diff
@@ -66,6 +67,19 @@ public static SeekableInputStream wrap(FSDataInputStream stream) {
     }
   }
 
+  private static boolean isWrappedStreamByteBufferReadable(FSDataInputStream stream) {
+    InputStream wrapped = stream.getWrappedStream();
+    if (wrapped == stream) {
+      throw new ParquetDecodingException("Illegal FSDataInputStream as wrapped itself");
```

Comment:
this can't happen. the inner stream is set in the constructor, so cannot take the not-yet-constructed class as an argument...no need to worry about recursion.